consensus/safety-rules/src/safety_rules.rs

```diff
@@ -120,7 +120,7 @@ impl<T: Payload> SafetyRules<T> {
                 .set_waypoint(&Waypoint::new_epoch_boundary(ledger_info)?)?;
             self.persistent_storage.set_last_voted_round(0)?;
             self.persistent_storage.set_preferred_round(0)?;
-            self.persistent_storage.set_epoch(ledger_info.epoch() + 1)?;
+            self.persistent_storage.set_epoch(next_epoch)?;
```

Comment:
you can use epoch_info.epoch and skip the next_epoch var

Reply:
oh, this must have appeared after your code refactor, it wasn't available before :)